cpp/src/slice2cpp/Gen.cpp

```diff
@@ -216,15 +216,10 @@ namespace
         }
         else
         {
-            throws.sort();
-            throws.unique();
-
-            //
             // Arrange exceptions into most-derived to least-derived order. If we don't
             // do this, a base exception handler can appear before a derived exception
             // handler, causing compiler warnings and resulting in the base exception
             // being marshaled instead of the derived exception.
-            //
             throws.sort(Slice::DerivedToBaseCompare());
```

Comment:
Note that in the languages, we _do_ still sort them by derived-to-base.
This sorting is stable (it won't re-arrange elements that are 'equal'). And this is necessary to prevent dead catch blocks.

We do this in all compilers except 'ice2slice' (where it's correct not to) and 'slice2php', 'slice2py', and 'slice2rb'.
It is suspicious that we aren't sorting exceptions in those languages.